pkg/api/metrics.go

```diff
@@ -78,16 +79,29 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext, reqDTO dtos.MetricReq
 		return response.Error(http.StatusInternalServerError, "Metric request error", err)
 	}
 
+	// This is insanity... but ¯\_(ツ)_/¯, the current query path looks like:
+	//  encodeJson( decodeBase64( encodeBase64( decodeArrow( encodeArrow(frame)) ) )
+	// this will soon change to a more direct route
+	qdr, err := resp.ToBackendDataResponse()
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "error converting results", err)
+	}
+	return toMacronResponse(qdr, nil)
+}
+
+func toMacronResponse(qdr *backend.QueryDataResponse, err error) response.Response {
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "error running query", err)
+	}
+
 	statusCode := http.StatusOK
-	for _, res := range resp.Results {
+	for _, res := range qdr.Responses {
 		if res.Error != nil {
-			res.ErrorString = res.Error.Error()
-			resp.Message = res.ErrorString
```

Comment:
The error is now part of the response object:
https://github.com/grafana/grafana-plugin-sdk-go/blob/master/backend/data.go#L104

*message* is missing, it is one of the key differences between the tsdb and sdk formats.  It has not been an issue for anything yet, but it could be added if necessary